server/db/models/game.js

```diff
@@ -55,398 +63,223 @@ var laserBlockedBy = {
-        direction: p.compassDirection
-      }))
-    });
-    return Promise.all(fired);
+  var game = this;
+  players.forEach(function(p){
+    fireOneLaser({
+      start: [p.position[0]+ p.bearing[0], p.position[1]+p.bearing[1]],
+      qty: 1,
+      bearing: p.bearing,
+    })
   });
 };
 
 gameSchema.methods.fireBoardLasers = function (){
-  var fired = [];
   for (var i = 0; i < 12; i ++){
-    fired.push(this.fireLasersInCol(i));
+    this.fireLasersInCol(i);
   }
-  return Promise.all(fired)
 }
 
 gameSchema.methods.fireLasersInCol = function (col) {
   var colStr = 'col' + col.toString();
-  return Board.findById(this.board).bind(this)
-  .then(function(b){
-    b.colStr.forEach(function(tile, i){
-      if(tile.edgeN.slice(0,4) === 'wall') {
-        return this.fireOneLaser({start: [i, col], qty:Number(tile.edgeN[4]), bearing: [1, 0], direction: 'S'});
-      }
-      if(tile.edgeE.slice(0,4) === 'wall') {
-        return this.fireOneLaser({start: [i, col], qty:Number(tile.edgeE[4]), bearing: [0, -1], direction: 'W'});
-      }
-      if(tile.edgeS.slice(0,4) === 'wall') {
-        return this.fireOneLaser({start: [i, col], qty:Number(tile.edgeS[4]), bearing: [-1, 0], direction: 'N'});
-      }
-      if(tile.edgeW.slice(0,4) === 'wall') {
-        return this.fireOneLaser({start: [i, col], qty:Number(tile.edgeW[4]), bearing: [0, 1], direction: 'E'});
-      }
-    });
+  this.board[colStr].forEach(function(tile){
+    if(tile.edgeN.slice(0,4) === 'wall') {
+      this.fireOneLaser({start: [i, col], qty:Number(tile.edgeN[4]), bearing: [1, 0, 'S']});
+    }
+    if(tile.edgeE.slice(0,4) === 'wall') {
+      this.fireOneLaser({start: [i, col], qty:Number(tile.edgeE[4]), bearing: [0, -1, 'W']});
+    }
+    if(tile.edgeS.slice(0,4) === 'wall') {
+      this.fireOneLaser({start: [i, col], qty:Number(tile.edgeS[4]), bearing: [-1, 0, 'N']});
+    }
+    if(tile.edgeW.slice(0,4) === 'wall') {
+      this.fireOneLaser({start: [i, col], qty:Number(tile.edgeW[4]), bearing: [0, 1, 'E']});
+    }
   })
 }
 
 gameSchema.methods.fireOneLaser = function(laser){
-  // laser has qty, bearing, direction (string), start properties
-  var currLoc = laser.start;
   var nextLoc = laser.start;
-  // check current location for obstacles to exiting
-  return this.getPlayerAt(currLoc).bind(this)
-  .then(function(p){
-    if (p){
-      return p.applyDamage(laser.qty);
-    }else{
-      return this.getTileAt(currLoc)
-      .then(function(t){
-        if(t[laserBlockedBy[laser.direction].exit]) return;
-        return t;
-      })
-    }
-  })
-  .then(function(t){
-    if(t) { //check ability to enter next tile
-      nextLoc[0] += laser.bearing[0];
-      nextLoc[1] += laser.bearing[1];
-      return this.getTileAt(nextLoc)
-      .then(function(t){
-        if(t && t[laserBlockedBy[laser.direction].enter]) return;
-        return t;
-      })
-    }
+  var tile, p;
+
+   //find a player at the current location
+  p = this.getPlayerAt(currLoc);
```

Comment:
Where is currLoc defined? I think it may have been deleted. Should it be laser.start?